### ISSUE TITLE
.Coffee extension on generated .tmp-globalize-webpack/ file

### ISSUE THREAD

**Issue by brenwell** [rxaviers/globalize-webpack-plugin#16]
I am using Coffeescript and when I am using this library I am having an issue that the generated file has the wrong extension. e.g.

**globalize-manager.coffee**

``` coffeescript
    Globalize = require('globalize')
    formatter = Globalize.currencyFormatter('EUR')
    formatter(amount)
```

produces a file named

**.tmp-globalize-webpack/...-globalize-manager.coffee**

Which is causing problems as it is a js file with the wrong extension. Any idea how to get around this?
For the moment I am using a js file instead.

Thanks again

**Comment by rxaviers** [comment]
It seems like the plugin keep the original extension, but by that time its content may have been converted to JS via coffee-loader. Is that the case? Should globalize plugin always use JS for the extension? I don't have an answer for your.

All I can say for now is that ideally we shouldn't be using a temporary file, but http://stackoverflow.com/questions/31147779/is-there-a-webpack-dependency-object-that-takes-content-instead-of-filepath-de

**Comment by brenwell** [comment]
>
It seems like the plugin keep the original extension, but by that time its content may have been converted to JS via coffee-loader. Is that the case?

Yes this appears to be the case

>
Should globalize plugin always use JS for the extension?

Seems like the simplest solution to me

Cheers

**Comment by arlair** [comment]
I think I ran into the same problem with TypeScript files.

**Comment by rxaviers** [comment]
Can you confirm if "Should globalize plugin always use JS for the extension?"? If so, I think this is pretty quick and easy. I'd be definetely open to accept this fix.

**Comment by arlair** [comment]
I am a bit confused about what is happening underneath and unfortunately cannot confirm.

**Comment by danwashusen** [comment]
Confirming that appending ".js" to the filepath fixes the issue:

```
GlobalizeCompilerHelper.prototype.createCompiledDataModule = function(request) {
  var filepath = this.getModuleFilepath(request) + ".js"; // module filepath doesn't take into account loaders (e.g. coffeescript)
  this.modules[filepath] = true;

  fs.writeFileSync(filepath, this.compile(this.developmentLocale, request));

  return filepath;
};
```

**Comment by rxaviers** [comment]
Fixed by #30
